statusio.go

```diff
@@ -15,14 +16,26 @@ import (
 type Client struct {
 	baseAddr string
 	client   *http.Client
+	scheme   string
+}
+
+// extractScheme detects if the url is http instead of https
+func extractScheme(rawurl string) (scheme, baseAddr string) {
+	u, err := url.Parse(rawurl)
+	if err != nil || u.Scheme == "" || u.Host == "" {
```

Comment:
this looks like it will hide an error on invalid urls.

Reply:
Next patch addresses this. Note that there is my statusio branch for bosun that goes with it because: 1- statusio is vendored in bosun, 2- NewClient() is returning an error now.